src/main/java/seedu/address/model/person/DateAbsentPredicate.java

```diff
@@ -0,0 +1,45 @@
+package seedu.address.model.person;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.function.Predicate;
+
+import seedu.address.commons.util.ToStringBuilder;
+
+/**
+ * Tests that a {@code Person}'s {@code attendance} does not contain the date
+ * provided.
+ */
+public class DateAbsentPredicate implements Predicate<Person> {
+    private final LocalDateTime classDate;
+
+    public DateAbsentPredicate(LocalDateTime classDate) {
+        this.classDate = classDate;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return new Attendance(false).equals(person.getAttendanceList().getMap().getOrDefault(classDate, null));
```

Comment:
Calling `getMap()` violates LoD. Instead, would it be better if we write a method like `getAttendance(LocalDateTime date)` inside `AttendanceList`?